.NET/Microsoft.Recognizers.Text.DataTypes.DataDrivenTests/TestTimexConvert.cs

```diff
@@ -15,276 +15,314 @@ public void DataTypes_Convert_CompleteDate()
         }
 
         [TestMethod]
-        public void DataTypes_Convert_Month()
+        [DataRow("XXXX-01", "January")]
+        [DataRow("XXXX-05", "May")]
+        [DataRow("XXXX-12", "December")]
+        public void DataTypes_Convert_Month(string timex, string expected)
         {
-            Assert.AreEqual("January", TimexConvert.ConvertTimexToString(new TimexProperty("XXXX-01")));
-            Assert.AreEqual("May", TimexConvert.ConvertTimexToString(new TimexProperty("XXXX-05")));
-            Assert.AreEqual("December", TimexConvert.ConvertTimexToString(new TimexProperty("XXXX-12")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_MonthAndYear()
+        [DataRow("2018-05", "May 2018")]
+        public void DataTypes_Convert_MonthAndYear(string timex, string expected)
         {
-            Assert.AreEqual("May 2018", TimexConvert.ConvertTimexToString(new TimexProperty("2018-05")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_WeekOfMonth()
+        [DataRow("XXXX-01-W01", "first week of January")]
+        [DataRow("XXXX-08-W03", "third week of August")]
+        public void DataTypes_Convert_WeekOfMonth(string timex, string expected)
         {
-            Assert.AreEqual("first week of January", TimexConvert.ConvertTimexToString(new TimexProperty("XXXX-01-W01")));
-            Assert.AreEqual("third week of August", TimexConvert.ConvertTimexToString(new TimexProperty("XXXX-08-W03")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_PartOfTheDay()
+        [DataRow("TDT", "daytime")]
+        [DataRow("TNI", "night")]
+        [DataRow("TMO", "morning")]
+        [DataRow("TAF", "afternoon")]
+        [DataRow("TEV", "evening")]
+        public void DataTypes_Convert_PartOfTheDay(string timex, string expected)
         {
-            Assert.AreEqual("daytime", TimexConvert.ConvertTimexToString(new TimexProperty("TDT")));
-            Assert.AreEqual("night", TimexConvert.ConvertTimexToString(new TimexProperty("TNI")));
-            Assert.AreEqual("morning", TimexConvert.ConvertTimexToString(new TimexProperty("TMO")));
-            Assert.AreEqual("afternoon", TimexConvert.ConvertTimexToString(new TimexProperty("TAF")));
-            Assert.AreEqual("evening", TimexConvert.ConvertTimexToString(new TimexProperty("TEV")));
+            Assert.AreEqual(expected, TimexConvert.ConvertTimexToString(new TimexProperty(timex)));
         }
 
         [TestMethod]
-        public void DataTypes_Convert_FridayEvening()
+        [DataRow("(XXXX-XX-XX,XXXX-XX-XX,P1D)", "1 day")]
+        public void DataTypes_Convert_Period(string timex, string expected)
```

Comment:
It's better to rename to DataTypes_Convert_Undefined_Period or DataTypes_Convert_Undefined_Range for clarity.